var/spack/repos/builtin/packages/kokkos/package.py

```diff
@@ -408,39 +406,35 @@ def setup_build_tests(self):
         self.cache_extra_test_sources(cmake_out_path)
         self.cmake_bin(set=True)
 
-    def build_tests(self, cmake_path):
+    def test_build(self):
         """Build test."""
-        cmake_bin = self.cmake_bin(set=False)
-
-        if not cmake_bin:
-            tty.msg("Skipping kokkos test: cmake_bin_path.txt not found")
-            return
+        # Skip if unsupported version
+        cmake_path = join_path(
+            self.test_suite.current_test_cache_dir, self.test_script_relative_path, "out"
+        )
 
-        cmake_args = [cmake_path, "-DEXECUTABLE_OUTPUT_PATH=" + cmake_path]
+        if not os.path.exists(cmake_path):
+            raise SkipTest(f"{cmake_path} is missing")
 
-        if not self.run_test(cmake_bin, options=cmake_args, purpose="Generate the Makefile"):
-            tty.warn("Skipping kokkos test: failed to generate Makefile")
-            return
+        cmake_bin = self.cmake_bin(set=False)
+        if not cmake_bin:
+            assert False, "cmake_bin_path.txt not found"
 
-        if not self.run_test("make", purpose="Build test software"):
-            tty.warn("Skipping kokkos test: failed to build test")
+        exe = which(cmake_bin)
+        exe(cmake_path, "-DEXECUTABLE_OUTPUT_PATH=" + cmake_path)
 
-    def run_tests(self, cmake_path):
-        """Run test."""
-        if not self.run_test(
-            "make", options=[cmake_path, "test"], purpose="Checking ability to execute."
-        ):
-            tty.warn("Failed to run kokkos test")
+        make = which("make")
+        make()
```

Comment:
The options that should be passed to `make()` are `cmake_path, "test"` as shown on old line 431.